packages/typespec-go/test/cadlranch/parameters/spreadgroup/zz_spreadalias_client.go

```diff
@@ -114,17 +114,123 @@ func (client *SpreadAliasClient) spreadAsRequestParameterCreateRequest(ctx conte
+	var err error
+	const operationName = "SpreadAliasClient.SpreadParameterWithInnerModel"
+	ctx = context.WithValue(ctx, runtime.CtxAPINameKey{}, operationName)
+	ctx, endSpan := runtime.StartSpan(ctx, operationName, client.internal.Tracer(), nil)
+	defer func() { endSpan(err) }()
+	req, err := client.spreadParameterWithInnerModelCreateRequest(ctx, id, name, xMSTestHeader, options)
+	if err != nil {
+		return SpreadAliasClientSpreadParameterWithInnerModelResponse{}, err
+	}
+	httpResp, err := client.internal.Pipeline().Do(req)
+	if err != nil {
+		return SpreadAliasClientSpreadParameterWithInnerModelResponse{}, err
+	}
+	if !runtime.HasStatusCode(httpResp, http.StatusNoContent) {
+		err = runtime.NewResponseError(httpResp)
+		return SpreadAliasClientSpreadParameterWithInnerModelResponse{}, err
+	}
+	return SpreadAliasClientSpreadParameterWithInnerModelResponse{}, nil
+}
+
+// spreadParameterWithInnerModelCreateRequest creates the SpreadParameterWithInnerModel request.
+func (client *SpreadAliasClient) spreadParameterWithInnerModelCreateRequest(ctx context.Context, id string, name string, xMSTestHeader string, _ *SpreadAliasClientSpreadParameterWithInnerModelOptions) (*policy.Request, error) {
+	urlPath := "/parameters/spread/alias/inner-model-parameter/{id}"
+	if id == "" {
+		return nil, errors.New("parameter id cannot be empty")
+	}
+	urlPath = strings.ReplaceAll(urlPath, "{id}", url.PathEscape(id))
+	req, err := runtime.NewRequest(ctx, http.MethodPost, runtime.JoinPaths(host, urlPath))
+	if err != nil {
+		return nil, err
+	}
+	req.Raw().Header["Content-Type"] = []string{"application/json"}
+	req.Raw().Header["x-ms-test-header"] = []string{xMSTestHeader}
+	body := struct {
+		Name string `json:"name"`
+	}{
+		Name: name,
+	}
+	if err := runtime.MarshalAsJSON(req, body); err != nil {
+		return nil, err
+	}
+	return req, nil
+}
+
 // SpreadWithMultipleParameters -
 // If the operation fails it returns an *azcore.ResponseError type.
+//   - requiredString - required string
+//   - optionalInt - optional int
+//   - requiredIntList - required int
+//   - optionalStringList - optional string
 //   - options - SpreadAliasClientSpreadWithMultipleParametersOptions contains the optional parameters for the SpreadAliasClient.SpreadWithMultipleParameters
 //     method.
-func (client *SpreadAliasClient) SpreadWithMultipleParameters(ctx context.Context, id string, xMSTestHeader string, prop1 string, prop2 string, prop3 string, prop4 string, prop5 string, prop6 string, options *SpreadAliasClientSpreadWithMultipleParametersOptions) (SpreadAliasClientSpreadWithMultipleParametersResponse, error) {
+func (client *SpreadAliasClient) SpreadWithMultipleParameters(ctx context.Context, id string, xMSTestHeader string, requiredString string, optionalInt int32, requiredIntList []int32, optionalStringList []string, options *SpreadAliasClientSpreadWithMultipleParametersOptions) (SpreadAliasClientSpreadWithMultipleParametersResponse, error) {
```

Comment:
change from cadl-ranch test change

Reply:
Hmm, I think the optional params should be placed in the options type.

Reply:
Looks like tcgc doesn't properly propagate the optional value for these params. e.g. for `optionalInt` I see it set to `optional: false`.

Reply:
Opened https://github.com/Azure/typespec-azure/issues/1218 to track it.

Reply:
> Looks like tcgc doesn't properly propagate the optional value for these params. e.g. for `optionalInt` I see it set to `optional: false`.

If that's true then we should NOT take the PR until that is fixed.

Reply:
No need to gate on this as it's not a regression. In addition, we don't properly handle this in the code generator either, so a fix is required on both sides.

Reply:
from my debug, the optional is true in tcgc model, am i miss sth?
![Uploading image.png…]()